ZSSRforKernelGAN/ZSSR.py

```diff
@@ -137,34 +145,12 @@ def set_disc_loss(self, D, DiscLoss):
     def run(self):
         # Run gradually on all scale factors (if only one jump then this loop only happens once)
         print('*' * 60 + '\nSTARTED ZSSR on: \"%s\"...' % self.input_img_path)
-        for self.sf_ind, (sf, self.kernel) in enumerate(zip(self.conf.scale_factors, self.kernels)):
-            # Relative_sf (used when base change is enabled. this is when input is the output of some previous scale)
-            sf = [sf, sf] if np.isscalar(sf) else sf
-            self.sf = np.array(sf) / np.array(self.base_sf)
-
-            self.output_shape = np.uint(np.ceil(np.array(self.input.shape[0:2]) * sf))
-
-            # Initialize network
-            self.init_sess(init_weights=self.conf.init_net_for_each_sf)
```

Comment:
what is it and why we can remove it?
init_sess

Reply:
In the previous implementation we had two calls for init_sess, one at the __init__ function of ZSSR and a second for each scale factor we had, so actually whenever there is only one scale factor the second call to init_sess is not needed, just time-consuming...

Now we have one call to init_sess at line 106.
But actually, this function is not needed anymore, since it is called only from __init__ function, we can just take its context outside and remove this function (notice it only dose initialization of class attributes and weights parameters)